browser/ui/sidebar/sidebar_controller.cc

```diff
@@ -76,17 +85,47 @@ bool SidebarController::DoesBrowserHaveOpenedTabForItem(
 
 void SidebarController::ActivateItemAt(absl::optional<size_t> index,
                                        WindowOpenDisposition disposition) {
+  SidebarTabHelper* helper = nullptr;
+  if (auto* active_tab = browser_->tab_strip_model()->GetActiveWebContents()) {
+    helper = SidebarTabHelper::FromWebContents(active_tab);
+  }
+  DVLOG(1) << __func__ << " index: " << index.value_or(-1u)
+           << " helper: " << helper;
+
   // disengaged means there is no active item.
   if (!index) {
     sidebar_model_->SetActiveIndex(index);
+    browser_active_panel_type_ = absl::nullopt;
```

Comment:
When chat panel is closed and previously global tab is also cleared here.

1) activate bookmark panel at tab 1
2) move to tab 2 and activate chat panel
3) close chat panel at tab 2
4) move to tab 1

I expected that bookmark panel is opened again when moved to tab 1 but it's not because of this line.
Is this intended behavior?

Reply:
I think it's debatable. Closing the panel can be seen as a global choice. Likewise, if I select a different "global" panel instead of chat (e.g. Reading List), then that will now become the active global panel. But I see your point. Really, this is a prototype to go along with the prototype AI Chat feature, so I don't know what feels right. Since you said you expected it to be otherwise, I'll change it...

Reply:
I don't have strong opinion about this. As this is prototype, I think we can start with current simple logic.
I was just curiou this is intended or not.

Reply:
Actually I tried it (at https://github.com/brave/brave-core/tree/sidebar-tab-specific-panel-temporary-close) and I don't think it feels right. The user could close it from the sidebar button in the browser window toolbar, and it feels bad to have the sidebar re-open when switching a tab. Also for the current tab, when closing a tab-specific panel, then switching to another tab and back again, the panel is open for that tab, it feels a little confusing.
Either way, I think we can revisit when we get feedback...